### PR TITLE
Support of trajectory aggregation for mrope multimodal model, and add multimodal prefix checks for trajectory merge

### PR DESCRIPTION
**Pull request by jackhu-bme**
## Summary
This PR enables trajectory aggregation for multimodal models using mRoPE, which was previously disallowed due to missing multimodal state and correctness concerns during aggregation.

## Motivation
Trajectory aggregation is an important feature for accelerating training in multi-turn agent reinforcement learning. Compared to transition-level aggregation, it processes an entire interaction trajectory as a single training sample, eliminating redundant computation on repeated history prefixes and significantly improving training throughput.

This aggregation strategy is commonly used in large language model pre-training and fine-tuning, and is described in more detail in the Agent Lightning documentation: https://agent-lightning.github.io/posts/trajectory_level_aggregation/

In the current implementation, trajectory aggregation is explicitly disabled for multimodal models using mRoPE via an assertion. This restriction was introduced because the aggregated trajectories did not carry sufficient multimodal state to guarantee correctness, which could lead to incorrect mRoPE position assignment when image inputs are involved.

This PR revisits that restriction by making the required multimodal information explicit and introducing additional validation to ensure correctness.

## Changes Made
- Removed the hard assertion that disabled trajectory aggregation for M-RoPE-based multimodal models, allowing trajectories to proceed when  image grid metadata is available.
- Propagated `image_grid_thw` during trajectory merges by deriving the multimodal grid metadata from the last merged trace and appending it to `image_grid_thw_list`, ensuring consistent `position_ids` computation and correct image token alignment.
- Added multimodal prefix consistency checks during trajectory aggregation: trajectory merges now require both token-level prefixes and image URL prefixes to match. Image URLs are semantically normalized (e.g., `file://` and `data:` URLs with identical content map to the same hash), and mismatches are logged for debugging.

## Testing
Not run with official examples, as the current repository does not yet include a multimodal, multi-turn trajectory example that exercises trajectory aggregation.

Existing examples fall into the following categories:
- Pure-text, multi-turn agents (e.g., conversational or planning demos), which are suitable for trajectory aggregation but do not involve multimodal inputs.
- Multimodal examples that are single-turn or stage-based pipelines, where images are provided per step but conversation history is not fed  back into subsequent turns, so trajectory-level prefix merging is not exercised.
- Single-shot or non-conversational tasks that do not involve trajectory aggregation.

The changes were verified locally using a custom multimodal, multi-turn workflow (X-ray image input with a crop tool), where conversation history is preserved across turns. Trajectory merges succeeded only when both token prefixes and image prefixes matched, and the new validation logic behaved as expected.

A standalone multimodal trajectory example (X-ray + crop tool) can be contributed in a follow-up PR if desired.

## Breaking Changes / Risks
- No breaking changes to public APIs.
- Trajectory aggregation for multimodal mRoPE models is now enabled under stricter validation. In rare cases where token prefixes match but image prefixes do not, aggregation will be skipped to preserve correctness.

### CODE REVIEW

**Comment by jackhu-bme** [comment]
@microsoft-github-policy-service agree

File: agentlightning/verl/daemon.py [hunk 1]
```diff
@@ -72,6 +72,63 @@ def _none_special_token_sequence(ids: List[int]) -> List[int]:
     return is_prefix, (template_mismatch, retoken_mismatch, others_mismatch)
 
 
+def _normalize_image_url(url: str) -> str:
+    if url.startswith("file://"):
+        path = url[len("file://") :]
+        try:
+            with open(path, "rb") as handle:
+                data = handle.read()
+            import hashlib
```
[the remark this answers is not in the document]
Comment (jackhu-bme) [reply agentlightning/verl/daemon.py:81]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/microsoft/agent-lightning/pull/469#discussion_r2740429603)

**Comment by ultmaster** [comment]
Please see whether the Copilot's comments make sense

**Comment by jackhu-bme** [comment]
> Please see whether the Copilot's comments make sense

Thanks, I will check out.
